### PR TITLE
SaraS project-todos

### PR DESCRIPTION
**Pull request by ssundblom**
Link to my deployed project:
https://ssundblom-todo.netlify.app/

My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/AddTask.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import styled from 'styled-components'
+
+import { tasks } from '../reducers/tasks'
+
+const AddTaskContainer = styled.form`
```
Comment (liza-minican) [inline code/src/components/AddTask.js:7]: Hey Sara, it is cool that you decided to use styled components and this one looks really neat!

File: code/src/components/AddTask.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+padding: 10px;
+height: 40px;
+`
+
+const TaskInput = styled.input.attrs({ type: 'text' })`
+background-color: #F6D1E5;
+border: none; 
+flex-grow: 1;
+padding: 10px;
+&:focus {
```
Comment (liza-minican) [inline code/src/components/AddTask.js:18]: Nice usage of focus in a styled component.

File: code/src/components/AddTask.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export const AddTask = () => {
+  const [task, setTask] = useState('')
+
+  const dispatch = useDispatch()
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    dispatch(tasks.actions.addItem({ task }))
+    setTask('')
```
Comment (liza-minican) [inline code/src/components/AddTask.js:42]: It is nice that you clear up the input when you are done.

File: code/src/components/CheckboxStyled.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+
+const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
```
Comment (liza-minican) [inline code/src/components/CheckboxStyled.js:4]: It is impressive that you used checkboxes, not an easy thing to style! Good job!

File: code/src/components/CheckboxStyled.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+overflow: hidden;
+padding: absolute;
+white-space: nowrap;
+`
+
+const StyledCheckbox = styled.div` 
+  display: inline-block;
+  width: 18px;
+  height: 18px;
+  background: ${props => props.checked ? '#4D1C39' : '#9A6A80'};
```
Comment (liza-minican) [inline code/src/components/CheckboxStyled.js]: Props inside style components is another thing I am very curious about and want to learn well. I like to see it in your project and learn from it:)

File: code/src/components/CheckboxStyled.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export const CheckboxStyled = ({ checked, onChange }) => {
+  const [isChecked, setChecked] = useState(checked)
+
+  const handleOnChange = event => {
+    setChecked(event.target.checked)
+    onChange()
+  }
+
+  const Checkbox = ({ className, checked, ...props }) => (
```
Comment (liza-minican) [inline code/src/components/CheckboxStyled.js:38]: What is the usage for this ...props, is it an array? just curious:)

File: code/src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { TaskCounter } from './TaskCounter'
+
+const HeaderContainer = styled.div`
+  padding: 15px;
+`
+
+const TodaysDate = styled.div`
```
Comment (liza-minican) [inline code/src/components/Header.js:10]: Do you need this empty styled component?

File: code/src/components/Task.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+const DeleteButton = styled.button`
+  margin-left: auto;
+  background-color: #F6D1E7;
+  border: none;
+  color: #B4889E;
+`
+
+const TaskText = styled.div`
+  text-decoration: ${props => props.checked ? 'line-through' : 'none'};
```
Comment (liza-minican) [inline code/src/components/Task.js:28]: I love the idea of crossing out the task that is done already 👍

File: code/src/components/TaskCounter.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+export const TaskCounter = () => {
+  const todos = useSelector(store => store.tasks.todos)
+
+  return (
+    <div>
+    <div> done: {todos.filter((task) => task.complete === true).length} </div>
```
Comment (liza-minican) [inline code/src/components/TaskCounter.js:9]: It is interesting to see a different approach for me as you use the filter method right here.

**Review by liza-minican — APPROVED** [review]
Great job, Sara!! You should be proud!! Smart logic and neat styled components!👍

File: code/src/components/AddTask.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+`
+
+const PlusButton = styled.button`
+background-color: #F6D1E5;
+border: none; 
+margin-left: auto;
+font-size: 18px;
+color: #6E355C;
+padding: 10px;
+`
```
Comment (HIPPIEKICK) [inline code/src/components/AddTask.js]: Indent all styled components:
```
const PlusButton = styled.button`
  background-color: #F6D1E5;
  border: none; 
  margin-left: auto;
  font-size: 18px;
  color: #6E355C;
  padding: 10px;
`
```

File: code/src/components/CheckboxStyled.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    <CheckboxContainer>
+      <HiddenCheckbox checked={checked} {...props}></HiddenCheckbox>
+      <StyledCheckbox checked={checked}> </StyledCheckbox>
```
Comment (HIPPIEKICK) [inline code/src/components/CheckboxStyled.js:41]: Nice!

File: code/src/components/AddTask.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    setTask('')
+  }
+
+  return (
+    <AddTaskContainer onSubmit={handleSubmit}>
+      <TaskInput
+        type='text'
+        placeholder="Add task"
+        value={task}
+        onChange={(event) => setTask(event.target.value)} />
```
Comment (HIPPIEKICK) [inline code/src/components/AddTask.js:51]: Prevent the user to submit empty todos and you're good to go :)

**Pull request closed without merging**